File: hello-typescript/src/server/protocol.ts
/** Protocol command types. */
export type CommandType = 'STREAM' | 'DATA' | 'QUERY';

/** Stream lifecycle commands (client→server). */
export type StreamCommand = 'CREATE' | 'COMPLETE' | 'CLOSE';

/** Data commands. */
export type DataCommand = 'READ';

/** Query commands. */
export type QueryCommand = 'GET_STATUS' | 'LIST_STREAMS';

/** Parsed command routing info. */
export interface CommandInfo {
  cmdType: CommandType;
  streamCmd?: StreamCommand;
  dataCmd?: DataCommand;
  queryCmd?: QueryCommand;
}

File: hello-rust/src/server/protocol.rs
/// Command type categories.
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum CommandType {
    Stream,
    Data,
    Query,
}

/// Stream lifecycle commands (client→server).
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum StreamCommand {
    Create,
    Complete,
    Close,
}

/// Data commands (client→server).
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum DataCommand {
    Read,
}

/// Query commands (client→server).
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum QueryCommand {
    GetStatus,
    ListStreams,
}

/// Parsed command routing info.
pub struct CommandInfo {
    pub cmd_type: CommandType,
    pub stream_cmd: Option<StreamCommand>,
    pub data_cmd: Option<DataCommand>,
    pub query_cmd: Option<QueryCommand>,
}
